pkg/rancher-components/src/components/Card/Card.vue

```diff
@@ -50,12 +51,37 @@ export default defineComponent({
       type:    Boolean,
       default: false,
     },
-  }
+    triggerFocusTrap: {
+      type:    Boolean,
+      default: false,
+    },
+  },
+  data() {
+    return { focusTrapInstance: {} as FocusTrap };
+  },
+  mounted() {
+    if (this.triggerFocusTrap) {
+      this.focusTrapInstance = createFocusTrap(this.$refs.cardContainer as HTMLElement, {
+        escapeDeactivates: true,
+        allowOutsideClick: true,
+      });
+
+      this.$nextTick(() => {
+        this.focusTrapInstance.activate();
+      });
+    }
+  },
+  beforeUnmount() {
+    if (this.focusTrapInstance && this.triggerFocusTrap) {
+      this.focusTrapInstance.deactivate();
+    }
+  },
```

Comment:
I have a hunch that this is something that we can reuse across different components. Have we considered if this would fit into a composable?

Reply:
We had an OOB discussion on how to approach this. The change requested does not need to be addressed at this moment, as we only have one instance of this pattern. If we see more, we will attempt to refactor in order to reduce duplication of logic.

For posterity, composition api features can be implemented in components that use the options API via the `setup()` option.